README.md

```diff
@@ -1,10 +1,82 @@
 [![Build Status](https://travis-ci.org/kubernetes-csi/driver-registrar.svg?branch=master)](https://travis-ci.org/kubernetes-csi/driver-registrar)
 # Cluster Driver Registrar
 
-A sidecar container that
+The cluster-driver-registrar is a sidecar container that creates a cluster-level
+[CSIDriver object](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/csi-api/pkg/crd/manifests/csidriver.yaml)
+for the CSI driver.
+
+This sidecar container is only needed if you need one of the following Kubernetes
+features:
+
+* Skip attach: For drivers that don't support ControllerPublishVolume, this
+  eliminates the need to deploy the external-attacher sidecar
+* Pod info on mount: This passes Kubernetes metadata such as Pod name and
+  namespace to the NodePublish call
 
 1. Registers the containerized CSI driver with Kubernetes and create a CSIDriver object
```

Comment:
We can remove this line. 

Also let's add `If you are not using one of these features, this sidecar container (and the creation of the [CSIDriver Object](csi-driver-object.md)) is not required. However, it is still recommended, because the [CSIDriver Object](csi-driver-object.md) makes it easier for users to easily discover the CSI drivers installed on their clusters.`